doc/cug.tex

```diff
@@ -4551,6 +4489,380 @@ \subsubsection{Collapsible Families In Suite Graphs}
+
+A negative offset notation \lstinline@<param_name=-1>@ is interpreted as the
+previous value in the ordered list of parameter values (integer or string).
+For example, to split a model run into multiple chunks with each chunk
+depending on the previous one, this graph:
+\begin{lstlisting}
+    graph = "model<run-1> => model<run>"  # for run = 1, 2, 3
+\end{lstlisting}
+expands to this:
+\begin{lstlisting}
+    model_run1 => model_run2 => model_run3
+\end{lstlisting}
+and this graph:
+\begin{lstlisting}
+    graph = "proc<size-1> => proc<size>"  # for size = small, big, huge
+\end{lstlisting}
+expands to this:
+\begin{lstlisting}
+    graph = "proc_small => proc_big => proc_huge"
+\end{lstlisting}
+
+\subsubsection{Task Families And Parameterization}
+
+Family names can be parameterized just like task names.  For instance, to
+define a family for tasks that share common values of a parameter,
+\begin{lstlisting}
+[runtime]
+    [[RUN<r>]]
+    [[model<r>]]
+        inherit = RUN<r>
+    [[post_proc<r>]]
+        inherit = RUN<r>
+# result: RUN_r1 contains model_r1 and post_proc_r1, and so on.
+\end{lstlisting}
+
+As described in Section~\ref{FamilyTriggers} family names can be used to
+trigger all members at once,
+\begin{lstlisting}
+    graph = "foo => FAMILY"
+\end{lstlisting}
+or to trigger another task off {\em all} members succeeding (or failing etc.):
+\begin{lstlisting}
+    graph = "FAMILY:succeed-all => foo"
+\end{lstlisting}
+or to trigger another task off {\em any} members succeeding (or failing etc.):
+\begin{lstlisting}
+    graph = "FAMILY:succeed-any => foo"
+\end{lstlisting}
+
+If the members of \lstinline=FAMILY= were generated with parameters such as
+\lstinline=member<m>=, you can just as easily trigger all members at once using
+parameter notation.
+\begin{lstlisting}
+    graph = "foo => member<m>"
+\end{lstlisting}
+Similarly, to trigger another task off {\em all} members succeeding (or failing
+etc.):
+\begin{lstlisting}
+    graph = "member<m> => foo"
+\end{lstlisting}
+
+Family names are still needed in the graph, however, to succinctly express
+``succeed-any'' triggering semantics, and all-to-all or any-to-all triggering:
+\begin{lstlisting}
+    graph = "FAM1:succeed-all => FAM2"
+\end{lstlisting}
+(Direct all-to-all and any-to-all family triggering is not recommended for
+efficiency reasons  though - see Section~\ref{EfficientInterFamilyTriggering}).
+
+For family {\em member-to-member} triggering just use parameterized members
+on each side of the arrow.  For example, if
+family \lstinline=OBS_GET= has members \lstinline=get<obs>= and
+family \lstinline=OBS_PROC= has members \lstinline=proc<obs>= then this,
+\begin{lstlisting}
+    graph = "get<obs> => proc<obs>"  # for obs = ship, buoy, plane
+\end{lstlisting}
+expands to this,
+\begin{lstlisting}
+    get_ship => proc_ship
+    get_buoy => proc_buoy
+    get_plane => proc_plane
+\end{lstlisting}
+
+\subsubsection{Static Chunking vs Dynamic Cycling}
```

Comment:
I guess I find this section a bit misleading tbh. Going in to it I was expecting some new form of dynamic cycling but what we really have is an R1 with a list of N entries where the task has to manipulate the nth parameter to generate a datetime. (I was also half expecting this to be dynamic sets of values e.g. at one cycle `<param>` evals as 1..3 whereas at another it evals as 1..10)